### ISSUE TITLE
If c-ares is installed manually, it will be used instead of third_party/c-ares

### ISSUE THREAD

**Issue by ncteisen** [grpc/grpc#11296]
### What operating system (Linux, Windows, …) and version?
 
MacOS. (might also occur on Linux)
 
### What did you do?

(assuming c-ares has been manually installed with apt-get)
make
 
### What did you expect to see?
make should have succeeded
 
### What did you see instead?
make failed with:
```
src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.c:65:30: error: field ‘dns_server_addr’ has incomplete type
   struct ares_addr_port_node dns_server_addr;
                              ^
src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.c: In function ‘grpc_dns_lookup_ares’:
src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.c:268:5: error: implicit declaration of function ‘ares_set_servers_ports’ [-Werror=implicit-function-declaration]
     int status = ares_set_servers_ports(*channel, &r->dns_server_addr);
```

**Comment by ncteisen** [comment]
Note that to fix this for now, you can run `sudo apt-get remove libc-ares-dev`

**Comment by ekarlso** [comment]
Any news on this?

**Comment by hsaliak** [comment]
Confirmed that this occurs on linux and that the workaround works..

**Comment by y-zeng** [comment]
Fixed by #11397
